src/encoder.rs

```diff
@@ -139,7 +142,7 @@ impl<W: Write> Writer<W> {
             let message = format!("wrong data size, expected {} got {}", data_size, data.len());
             return Err(EncodingError::Format(message.into()));
         }
-        let mut zlib = deflate::write::ZlibEncoder::new(Vec::new(), self.info.compression);
+        let mut zlib = deflate::write::ZlibEncoder::new(Vec::new(), self.info.compression.clone());
```

Comment:
I think it's possible to avoid using clone but I thought the convenience is worth it, since:

1. It's only a once-off initialisation cost.
2. A tagged enum variant with no internal data is basically just an integer (I would assume), so the allocation is tiny.

Reply:
I wouldn't worry about it too much for any static size.

Reply:
Bit late now, but if you had also derived Copy the clone call wouldn't be needed. (The generated code will be the same either way)

Reply:
`Copy` is a commitment that it will stay `Copy`. I'm not sure.